### PR TITLE
Add support so subsitutions can use JsonPath

### PR DESCRIPTION
**Pull request by BlythMeister**
Use the standard subsitution files to update JSON rather than XML.

This will only support ChangeValue & ReplacementContent (for now)

Adding new and add child content are out of scope for now.

### CODE REVIEW

**Comment by BlythMeister** [comment]
@richard-green changes made as requested

File: src/Ensconce.Update.Tests/Ensconce.Update.Tests.csproj [hunk 1]
```diff
@@ -36,6 +36,11 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="Microsoft.CSharp" />
+    <Reference Include="Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\packages\Newtonsoft.Json\lib\net45\Newtonsoft.Json.dll</HintPath>
```
Comment (richard-green) [inline src/Ensconce.Update.Tests/Ensconce.Update.Tests.csproj]: I think this should be a paket reference?

File: src/Ensconce.Update/UpdateFile.cs [hunk 2]
```diff
@@ -244,85 +289,172 @@ private static void ReplaceChildNodes(Lazy<TagDictionary> tagValues, XContainer
...
         }
 
-        private static Substitution BuildSubstitions(XElement change, IXmlNamespaceResolver nsm, Lazy<TagDictionary> tagValues)
+        private static string UpdateJson(Lazy<TagDictionary> tagValues, IEnumerable<Substitution> subs, JObject baseJson, IXmlNamespaceResolver nsm, XNode subsXml)
+        {
+            foreach (var sub in subs.Where(x => x.Execute))
+            {
+                Logging.Log($"Updating JsonPath {sub.Path}");
+
+                var xPathMatches = baseJson.SelectTokens(sub.Path.RenderTemplate(tagValues)).ToList();
```
Comment (richard-green) [inline src/Ensconce.Update/UpdateFile.cs]: => pathMatches

File: src/Ensconce.Update/UpdateFile.cs [hunk 3]
```diff
@@ -244,85 +289,172 @@ private static void ReplaceChildNodes(Lazy<TagDictionary> tagValues, XContainer
...
+
+                if (xPathMatches.Count == 0)
+                {
+                    throw new ApplicationException($"JsonPath select of {sub.Path} returned no matches.");
+                }
+
+                if (!sub.PathMatchAll && xPathMatches.Count > 1)
+                {
+                    throw new ApplicationException($"JsonPath select of {sub.Path} returned multiple matches. If the intention was to update all matches, use the attribute matchAll=\"true\" on the XPath or Change node.");
```
Comment (richard-green) [inline src/Ensconce.Update/UpdateFile.cs]: XPath => jsonPath?

**Pull request merged**